### PR TITLE
Adds parsed IO queue

### PR DESCRIPTION
**Pull request by mariuszbarczak**
Signed-off-by: Mariusz Barczak <mariusz.barczak@intel.com>

### CODE REVIEW

**Comment by tomaszrybicki** [comment]
retest this please

**Comment by mariuszbarczak** [comment]
retest this please

**Comment by mariuszbarczak** [comment]
retest this please

File: source/octf/trace/parser/ParsedIoTraceEventQueue.h [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+     * @brief Gets parsed event on queue front
+     * @return Parsed event on queue front
+     */
+    const proto::trace::ParsedEvent &front();
+
+    /**
+     * @brief Checks if queue is empty (reached end of the event trace)
+     *
+     * @retval true - queue empty
+     * @retval false - queue no empty
```
Comment (mmkayPL) [inline source/octf/trace/parser/ParsedIoTraceEventQueue.h]: queue not* empty

File: source/octf/trace/parser/ParsedIoTraceEventQueue.h [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+#include <octf/utils/NonCopyable.h>
+
+namespace octf {
+
+/**
+ * This class is a queue of parsed IO trace events.
+ *
+ * It parses a trace and allows to access parsed IOs using queue interface.
+ *
+ * @throws Exception when accessing empty queue (calling front or pop)
```
Comment (mmkayPL) [inline source/octf/trace/parser/ParsedIoTraceEventQueue.h]: It seems like the exception is thrown when event parsing had an exception, not when accessing an empty queue

File: tests/octf/trace/parser/ParsedIoTraceEventQueueTest.cpp [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    try {
+        SetupTestOutput(test_info_);
+
+        // Create trace
+        TestTrace trace(TRACE_LENGTH);
+        ParsedIoTraceEventQueue queue(trace.getTraceSummary().tracepath());
+
+        // Wait a time until queue will reach a limit,
+        // exit this branch to check if we destroy queue object properly
+        std::chrono::milliseconds sleppTime(1000);
```
Comment (mmkayPL) [inline tests/octf/trace/parser/ParsedIoTraceEventQueueTest.cpp]: sleep

**Pull request merged**